### ISSUE TITLE
[BUG] `Circuit.draw()` returns an error when gate parameter is a string

### ISSUE THREAD

**Issue by ValentinS4t1qbit** [goodchemistryco/Tangelo#388]
# Issue: Bug Report

We wanted to provide a default `draw` method on the `Circuit` class in Tangelo for convenience, and decided that `cirq` 's would do: when we call `draw()` we implicitly ask `cirq` to draw it for us (Thanks `cirq` ❤️ )

Tangelo circuits can take almost anything as a value for `gate.parameter`, including a string. `cirq` is however not happy with that:

```
circuit = Circuit([Gate('H', 0), Gate('CNOT', 1, 0), Gate('RY', 1, parameter='theta')])
circuit.draw()
```
<img width="1116" alt="Screenshot 2024-05-28 at 2 48 15 PM" src="https://github.com/goodchemistryco/Tangelo/assets/41597680/e6fcce6f-d2ed-46ea-bd58-2e2c0cdafdf0">

Can you help us ensure `draw()` works for strings as parameters ?

**How to win**

It turns out `cirq` is happy if we define a `sympy` symbol. Then we observe that both `draw` and `print` in Tangelo work  just fine:

<img width="860" alt="Screenshot 2024-05-28 at 3 08 03 PM" src="https://github.com/goodchemistryco/Tangelo/assets/41597680/fddc1176-1da6-4aa9-855b-8cc6fd50ec6b">

Which means one could ask `draw` to make a copy of the circuit on-the-fly (we don't want the method to modify the object itself) that replaces string values for parameters by equivalent sympy symbols.

- Try with an arbitrary circuit of 100 gates with parameters: if this takes less than 2-3 seconds, then your solution is performant enough. 
- If not, compare the time of your new `draw` function and and show us the difference in performance with the old one (which you can keep around and rename: we'll only push the new one into the PR).

**Validation**

- Verify that `draw` and `print` give satisfying results on the circuit above
- give `valentin.senicourt@sandboxquantum.com` access to a Google Colab notebook whose first cell installs Tangelo from your branch and then runs those draw / print cells with a few clicks and if it looks good, we're good !

**Comment by anushkrishnav** [comment]
Hey, I saw this issue on Unitary hacks , I think I can get it done, I will take this issue

**Comment by alexfleury-sb** [comment]
Thank you for looking into this! We will look forward updates, and provide additional info if needed.

**Comment by anushkrishnav** [comment]
Hopefully, with my PR, This issue has been completed. !!

**Comment by ValentinS4t1qbit** [comment]
Solved by @anushkrishnav in #391
